Backpack/TabBarController/Classes/BPKTabBarController.m

```diff
@@ -16,21 +16,37 @@
  * limitations under the License.
  */
 
-/*
- * Note that in some places we are having to resolve the correct UIColors from dynamic values ourselves to ensure
- * that we support overriding UIViewController interface styles. if we allow UIKit to resolve them for us,
- * it may do so wrongly.
- * This is because UIKit only sets the UIView's trait collection on a handful of lifecycle methods, and we
- * are using dynamic colours outside of these. If we do not manually resolve colours using self.traitCollection,
- * then UIKit will fall back to using [UITraitCollection currentTraitCollection] which does not reflect the override.
- */
-
 #import "BPKTabBarController.h"
 
+#import <Backpack/Color.h>
+
 NS_ASSUME_NONNULL_BEGIN
 
 @implementation BPKTabBarController
 
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        [self setup];
+    }
+    return self;
+}
+
+- (instancetype _Nullable)initWithCoder:(NSCoder *)coder
+{
+    self = [super initWithCoder:coder];
+    if (self) {
+        [self setup];
+    }
+    return self;
+}
+
+-(void)setup {
```

Comment:
This should be done in `viewDidLoad`. I suspect that using `self.tabBar` in the init like this will cause the view to load prematurely.

Reply:
Won't that override the colour if a consumer sets it first though?
```
let tabBarController = BPKTabBarController()
tabBarController.tabBar.tintColor = ...
// viewDidLoad happens
```

Reply:
As for using UIAppearance, shouldn't that be more of an app-level thing rather than a side-effect of the component? Might be hard to debug if our component is changing the UIAppearance?

Reply:
With `UIAppearance` we can have a central method to call which sets up all our styles, so no effect without calling it.

> Won't that override the colour if a consumer sets it first though?
There's will be later I think, in any case uisng `UIAppearance` would solve that issue

Reply:
Ah good shout!